test/widgets/action_sheet_test.dart

```diff
@@ -442,6 +445,44 @@ void main() {
             });
       });
 
+      testWidgets('on topic move, acts on new topic', (tester) async {
+        final stream = eg.stream();
+        const topic = 'old topic';
+        final message = eg.streamMessage(flags: [MessageFlag.read],
+          stream: stream, topic: topic);
+        await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+        // Get the action sheet fully deployed while the old narrow applies.
+        await tester.pumpAndSettle();
+
+        final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        final newStream = eg.stream();
+        const newTopic = 'other topic';
+        // This result isn't quite realistic for this request: it should get
+        // the updated channel/stream ID and topic, because we don't even
+        // start the request until after we get the move event.
+        // But constructing the right result is annoying at the moment, and
+        // it doesn't matter anyway: [MessageStoreImpl.reconcileMessages] will
+        // keep the version updated by the event.  If that somehow changes in
+        // some future refactor, it'll cause this test to fail.
+        connection.prepare(json: _getMessagesResult(message).toJson());
```

Comment:
For tests where we need to prepare the new messages, I [typically used](https://github.com/zulip/zulip-flutter/blob/ffa41be876743ee9b57982c7a315e6f97fe65491/test/model/message_list_test.dart#L926-L938) `eg.updateMessageEventMoveTo` with the message that has the new channel and topic, but I can see that we are in favor of `eg.updateMessageEventMoveFrom` here because in this context — moving away from the current narrow — the "move from" variant feels a bit more natural to use.